svgdigitizer.py

```diff
@@ -32,38 +43,83 @@ def get_points(self):
-        
-        for shape in ['circle', 'ellipse']:
-            for path in self.doc.getElementsByTagName(shape):
-                title = path.getElementsByTagName("title")
-                titletext = title[0].firstChild.data
-                position, value = titletext[:2], float(titletext[3:])
-                
-                ref_points[position] = [float(path.getAttribute('cx')), float(path.getAttribute('cy'))] 
-                real_points[position] = value
+
+
+        for text in self.doc.getElementsByTagName('text'):
+
+            # parse text content
+
+            text_content = text.firstChild.firstChild.data
+            regex_match =re.match(ref_point_regex_str, text_content)
+            if regex_match:
+                ref_point_id = regex_match.group("point")
+                real_points[ref_point_id] = float(regex_match.group("value"))
+                ref_points[ref_point_id] = {'x': float(text.firstChild.getAttribute('x')), 'y': float(text.firstChild.getAttribute('y'))}
+
         
         print('Ref points: ', ref_points)
         print('point values: ',real_points)
         return ref_points, real_points
         
+    def get_scalebars(self):
+        scalebars = {}
+
+        for path in self.doc.getElementsByTagName('path'):
+            try:
+                title = path.getElementsByTagName('title')[0].firstChild.data
+            except IndexError: # skip without title text
+                continue
+
+            regex_match = re.match(scalebar_regex_str, title)
+            if regex_match:
+                parsed_path = parse_path(path.getAttribute('d'))
+                scalebars[regex_match.group("axis")] = {}
+                if regex_match.group("axis") == 'x':
+                    scalebars[regex_match.group("axis")]['ref'] = abs(parsed_path.point(1).real - parsed_path.point(0).real)
+                elif regex_match.group("axis") == 'y':
+                    scalebars[regex_match.group("axis")]['ref'] = abs(parsed_path.point(1).imag - parsed_path.point(0).imag)
+                scalebars[regex_match.group("axis")]['real'] = float(regex_match.group("value"))
+
+        return scalebars
+
+    def get_scaling_factors(self):
+        scaling_factors = {'x': 1, 'y': 1}
+        for text in self.doc.getElementsByTagName('text'):
+
+            # parse text content
+            text_content = text.firstChild.firstChild.data
+            regex_match = re.match(scaling_factor_regex_str, text_content)
+
+            if regex_match:
+                print(regex_match.group("value"))
+                scaling_factors[regex_match.group("axis")] = float(regex_match.group("value"))
+        return scaling_factors
+
     def get_parsed(self):
         '''cuve function'''
         self.allresults = {}
         for pathid, pvals in self.figpaths.items():
             self.allresults[pathid] = self.get_real_values(pvals)
     
 
-    def get_trafo(self, xrefpath, xrefvalues):
+    def get_trafo(self, axis):
```

Comment:
What kind of transformations can happen here? I assume it could be any [linear affine transformation](https://en.wikipedia.org/wiki/Affine_transformation). Should we then maybe use a matrix to represent it?

Reply:
I don't know the mathematical term for the transformation but of course I'm in favor of a mathematical sound solution. Is this solution also suitable for waterfall(/3D) graphs? (will lookup an example)

Reply:
I think so. These transformations contain shears and I think that's what you need for waterfalls.